source/Sashimi.AzureServiceFabric/Endpoints/AzureServiceFabricClusterEndpointValidator.cs

```diff
@@ -0,0 +1,34 @@
+using FluentValidation;
+
+namespace Sashimi.AzureServiceFabric.Endpoints
+{
+    class AzureServiceFabricClusterEndpointValidator : AbstractValidator<AzureServiceFabricClusterEndpoint>
+    {
+        public AzureServiceFabricClusterEndpointValidator()
+        {
+            RuleFor(a => a.ConnectionEndpoint).NotEmpty();
+            RuleFor(a => a.SecurityMode).NotNull();
+            RuleFor(a => a.ServerCertThumbprint)
+                .NotEmpty()
+                .When(a => a.SecurityMode != AzureServiceFabricSecurityMode.Unsecure);
+            RuleFor(a => a.ClientCertVariable)
+                .NotEmpty()
+                .When(a => a.SecurityMode == AzureServiceFabricSecurityMode.SecureClientCertificate);
+            RuleFor(a => a.AadCredentialType)
```

Comment:
This used to be `.NotEmpty()` which seems to give you an error if the enum is set to its default value. I've made it more explicit.